tests/test_core.py

```diff
@@ -7953,7 +7953,6 @@ def setUp(self):
     os.chdir(self.get_dir()) # Ensure the directory exists and go there
 
     self.emcc_args = emcc_args[:]
-    Settings.load(self.emcc_args)
```

Comment:
Without this, we won't notice pre-existing changes to `self.emcc_args`, I think? E.g. in `asm2g` we set `ASSERTIONS` on via `emcc_args`.

(to make sure I'm not missing something, I diffed the Settings serialization before and after this call, and it does look different)

Reply:
I believe the ever since #6801, the contents of the Settings object in the test runner is not used.  If the tests themselves use `get_setting` and `set_setting` and then only those settings get serialized onto the emcc command line.   So anything you do the Settings object in the test process is not relevant.

Reply:
Actually you were right.. its little more subtle.  get_settings() has the return the right thing.  I update the test suites to be explicit about their settings.

Reply:
Hmm, I think there might be even more here - `-O1` and above will set `ASSERTIONS=0`. I think with the current code here, that would not be set?